homeassistant/helpers/device_registry.py

```diff
@@ -0,0 +1,74 @@
+"""Provide a way to connect entities belonging to one device.
+"""
+
+import logging
+
+import attr
+
+from homeassistant.core import callback
+from homeassistant.loader import bind_hass
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_REGISTRY = 'device_registry'
+
+
+@attr.s(slots=True, frozen=True)
+class DeviceEntry:
+    """Device Registry Entry."""
+
+    unique_id = attr.ib(type=str)
+    serial = attr.ib(type=str)
+    manufacturer = attr.ib(type=str)
+    model = attr.ib(type=str)
+    connection = attr.ib(type=tuple)
```

Comment:
I think the idea was for this to be a tuple of tuples. A device can have multiple connections. For example a Raspberry Pi has WiFi and LAN connection.

Reply:
Should we use some kind of standardisation for the types?

Reply:
```python
CONNECTION_ZIGBEE = 'zigbee'
CONNECTION_MAC = 'mac'
```

I guess we can also have duplicates (wifi + lan = 2 macs ?)

Reply:
I wonder how often you use both wifi and lan for a device that is connected to hass